### PR TITLE
Updated error messages for gpg based on if you have a device or not

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers this takes in a new param we get from the service and uses it to show different errors in the gpg provisioning flow

### CODE REVIEW

File: shared/engine/errors.js [hunk 1]
```diff
@@ -19,17 +19,15 @@ export default class EngineError extends Error {
     if (!err) {
       err = {}
     }
-    super(err.desc)
+
+    const niceFunc = niceMap[nameFix(err.name)]
+    super(niceFunc && niceFunc(err) || err.desc || err.code)
```
Comment (chrisnojima) [inline shared/engine/errors.js]: we need to pass this in the constructor

File: shared/engine/index.js [hunk 2]
```diff
@@ -290,7 +290,7 @@ class Engine {
 
       this.rpcClient.invoke(method, [param], (err, data) => {
         if (printRPC) {
-          logLocal('RPC ◀', method, param, err, data)
+          logLocal('RPC ◀', method, param, err, err && err.raw, JSON.stringify(data))
```
Comment (chrisnojima) [inline shared/engine/index.js]: else we can't actually see the raw values we get from the error

File: shared/login/register/error/index.render.desktop.js [hunk 3]
```diff
@@ -7,24 +7,52 @@ import {errorMap} from '../../../engine/errors'
 import openURL from '../../../util/open-url'
 
 const renderError = error => {
+  const fields = (error.raw && error.raw.fields || []).reduce((acc, f) => {
```
Comment (chrisnojima) [inline shared/login/register/error/index.render.desktop.js]: we get this back in a kinda funky way, should likely move this into the action or smart component but i think we'll need a larger ticket to clean this up as we have a lot of this kind of logic in 'dumb' components

File: shared/login/register/select-other-device/index.render.js.flow [hunk 4]
```diff
@@ -3,7 +3,7 @@
 import React from 'react'
 
 export type Props = {
-  devices: Array<{id: string, name: string, type: 'mobile' | 'desktop'}>,
+  devices: Array<{deviceID: string, name: string, type: 'mobile' | 'desktop'}>,
```
Comment (chrisnojima) [inline shared/login/register/select-other-device/index.render.js.flow]: really its deviceID

File: shared/login/register/select-other-device/index.render.desktop.js [hunk 5]
```diff
@@ -32,7 +32,7 @@ const Render = ({onBack, devices, onWont, onSelect}: Props) => (
     onBack={onBack}>
     <Text type='Header' style={styles.header}>Select a device to connect with:</Text>
     <div style={styles.devicesContainer}>
-      {devices.map(d => <Row onSelect={onSelect} {...d}/>)}
+      {devices.map(d => <Row onSelect={onSelect} {...d} key={d.deviceID}/>)}
```
Comment (chrisnojima) [inline shared/login/register/select-other-device/index.render.desktop.js]: need the key here else react complains (correctly)

**Comment by MarcoPolo** [comment]
How can I get into this state?

**Comment by chrisnojima** [comment]
Start provisioning your local w/ users that have the various states. so make an account w/ nothing, make an account with just a device key, make an account w/ gpg w/o uploading a key, make an account w/ uploading a key etc

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Comment by cjb** [comment]
:thumbsup:

**Pull request merged**